### DISCUSSION TITLE
Improved Type Annotation Display for Class and Type Alias

### DISCUSSION THREAD

**Discussion by Fuxuan-CN** [microsoft/pylance-release#8176] [Enhancement]
#### Description
It would be beneficial if VSCode could display whether a function's return type is a class or a type alias directly in the editor. Currently, when hovering over a function signature, it's not clear whether a type hint refers to a class or a type alias unless you look at the definition. Displaying this information as `(class) TypeName` or `(type) TypeName` would greatly enhance code readability and understanding.

#### Use Case
When working with complex type annotations, especially in large projects, it's often difficult to quickly discern whether a type hint is a class or a type alias. This feature would allow developers to quickly understand the nature of the type without navigating away from the current code context.

#### Expected Behavior
When hovering over a function signature or in the editor, if the return type is a class, display it as `(class) TypeName`. If it's a type alias, display it as `(type) TypeName`.

#### Current Behavior
The editor does not differentiate between class and type alias in type hints.

#### Examples
Here are some examples to illustrate the expected behavior:

```python
# Example 1: When the return type is a class
class Test:
    pass

def example() -> Test:
    # Function implementation
    pass
# Expected hover text: def example() -> (class) Test: ...

# Example 2: When the return type is a type alias
from typing import Tuple

type Test = Tuple[int, int]

def example() -> Test:
    # Function implementation
    pass
# Expected hover text: def example() -> (type) Test: ...

**Comment by rchiodo** [comment]
Transferring to a discussion item for up votes. This would be a UI change that might not be popular with everyone.
